custom/icds/management/commands/populate_mother_name.py

```diff
@@ -0,0 +1,212 @@
+                                updates.pop(case_id)
+                        for case_id, mother_name in updates.items():
+                            writer.writerow([case_id, mother_name])
+                        exceptions_raised = self._submit_update_form(updates, exceptions_raised)
+                        if self.log_progress:
+                            print("cases updated: %s" % counter)
+                        updates = {}
+                        counter = 0
+                # update the pending batch
+                for case_id, mother_name in updates.items():
+                    writer.writerow([case_id, mother_name])
+                exceptions_raised = self._submit_update_form(updates, exceptions_raised)
+
+    def _submit_update_form(self, updates, exceptions_raised):
+        update_case_blocks = self.create_case_blocks(updates)
+        if not update_case_blocks:
+            return exceptions_raised
+        for attempt in range(MAX_RESCUE_EXCEPTIONS_ON_UPDATE):
+            try:
+                submit_case_blocks(update_case_blocks, DOMAIN, user_id=SYSTEM_USER_ID)
+            except Exception as e:
+                exc = sys.exc_info()
+                exceptions_raised += 1
+                if self.log_progress:
+                    print("rescuing exception %s %s" % (exceptions_raised, str(e)))
+                if exceptions_raised > MAX_RESCUE_EXCEPTIONS_ON_UPDATE:
+                    six.reraise(*exc)
+                else:
+                    time.sleep(60)  # wait for 1 min before trying again
+            else:
+                break
+        return exceptions_raised
+
+    def create_case_blocks(self, updates):
+        case_blocks = []
+        for case_id, mother_name in updates.items():
+            case_block = CaseBlock(case_id,
+                                   update={MOTHER_NAME_PROPERTY: mother_name},
+                                   user_id=SYSTEM_USER_ID)
+            case_block = ElementTree.tostring(case_block.as_xml()).decode('utf-8')
+            case_blocks.append(case_block)
+        return case_blocks
+
+    def _reassured_case_ids_to_update(self, case_ids):
+        # reconfirm the cases before updating to avoid removing updates in between
+        # fetching case ids and updating
+        invalid_cases = self.case_accessor.get_cases(case_ids)
+        case_ids_list = []
+        for invalid_case in invalid_cases:
+            if self._case_needs_to_be_updated(invalid_case):
+                case_ids_list.append(invalid_case.case_id)
+        return case_ids_list
+
+
+def find_test_awc_locations():
+    test_locations = set()
+    for location in SQLLocation.active_objects.filter(location_type__code='state', domain=DOMAIN):
+        if location.metadata.get('is_test_location') == 'test':
+            test_locations.update(
+                location.get_descendants(include_self=True).
+                filter(location_type='awc').values_list('location_id', flat=True)
```

Comment:
confirming we dont need `location_type__code` filter here like above. just `location_type` will work?

Reply:
Thanks for catching that, it has to be `location_type__code`